pkg/helm/client/client.go

```diff
@@ -62,10 +54,32 @@ func (c *restClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
 }
 
 func (c *restClientGetter) ToRawKubeConfigLoader() clientcmd.ClientConfig {
+	return c.namespaceConfig
+}
+
+var _ clientcmd.ClientConfig = &namespaceClientConfig{}
+
+type namespaceClientConfig struct {
+	namespace string
+}
+
+func (c namespaceClientConfig) RawConfig() (clientcmdapi.Config, error) {
+	return clientcmdapi.Config{}, nil
+}
+
+func (c namespaceClientConfig) ClientConfig() (*rest.Config, error) {
+	return nil, nil
```

Comment:
Why it is required since always return nil and nil?

Reply:
These functions are required because `ToRawKubeConfigLoader()` needs to return a `clientcmd.ClientConfig` which is an interface that requires that all of these functions be implemented.

As far as I can tell, Helm only uses `ToRawKubeConfigLoader().Namespace()` and not the other functions (I'll double check this), so it should be okay for those to be no-ops. We _could_ have those unused functions return `fmt.Errorf("not implemented")` if that makes sense.

Reply:
Just following up on the question of how `ToRawKubeConfigLoader()` is used. Here's the grep output from the Helm project:

```console
$ grep -r ToRaw *
pkg/kube/factory.go:	// ToRawKubeConfigLoader return kubeconfig loader as-is
pkg/kube/factory.go:	ToRawKubeConfigLoader() clientcmd.ClientConfig
pkg/kube/client.go:	if ns, _, err := c.Factory.ToRawKubeConfigLoader().Namespace(); err == nil {
pkg/cli/environment.go:	if ns, _, err := s.RESTClientGetter().ToRawKubeConfigLoader().Namespace(); err == nil {
```

So yeah, only `Namespace()` actually needs to be implemented.